[Draft] [Discussion] Fix frontend conv_add_relu fusion issue

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97431
* __->__ #97122
* #97133
* #97418
* #97132
* #96475

**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```

leslie-fang-intel committed:
Fix frontend conv_add_relu fusion issue

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288


[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288


[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/match_utils.py:
```diff
@@ -12,7 +12,8 @@
 from ..utils import (
-    MatchAllNode
+    MatchAllNode,
+    conv_add_pattern_list
```
I feel we probably don't want to hack for special patterns in the general match_utils.py function..

leslie-fang-intel replied:
Thanks for the comment. Per your reply here https://github.com/pytorch/pytorch/issues/96288#issuecomment-1487488569, should we wait for the new programmable API ready to fix this issue?

jerryzh168 replied:
yes, it should be ready in 2 weeks

leslie-fang-intel replied:
That's great. Will check the fix after it's ready. Thanks~

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Draft] [Discussion] Fix frontend conv_add_relu fusion issue"


**Summary**
Fix the issue https://github.com/pytorch/pytorch/issues/97120 and https://github.com/pytorch/pytorch/issues/96288. The main change in this PR is: If a node is an extra input node of a `conv_add_relu` fusion pattern such as the `MatchALL` node below. We will skip add this `MatchALL` node into `match_map` [here](https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/match_utils.py#L184).
```
    Input
    /  \
Conv   MatchALL
    \   /
     ADD
      |
    ReLU
      |
   Output
```



[ghstack-poisoned]
